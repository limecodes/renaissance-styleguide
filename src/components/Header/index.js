import React from 'react'
import styled from 'styled-components'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

const HeaderWrapper = styled.div`
  width: 100%;
  border-bottom: 1px solid #000;
  background-color: #fff;
`

const Links = styled.ul`
  display: flex;
  flex-direction: row;
  margin: 0;
  padding: 0;
  list-style: none;
`

const Link = styled.li`
  display: inline-block;
  padding: 1rem;
`

const CloseButton = styled.button`
  border: 0;
  background: transparent;
`

export default () => (
  <HeaderWrapper>
    <Links>
      <Link>
        <CloseButton>
          <FontAwesomeIcon icon="bars" />
        </CloseButton>
      </Link>
      <Link>
        <a href="/home">Home</a>
      </Link>
      <Link>
        <a href="/patients">Patients</a>
      </Link>
    </Links>
  </HeaderWrapper>
)
