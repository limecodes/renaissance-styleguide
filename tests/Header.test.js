import React from 'react'
import renderer from 'react-test-renderer'
import { library } from '@fortawesome/fontawesome-svg-core'
import { faBars } from '@fortawesome/free-solid-svg-icons'

import Header from '../src/components/Header'

library.add(faBars)

test('Renders Correctly', () => {
  const tree = renderer.create(<Header />)

  expect(tree).toMatchSnapshot()
})
