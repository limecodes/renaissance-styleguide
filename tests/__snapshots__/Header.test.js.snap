// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Correctly 1`] = `
<div
  className="sc-bdnylx zjKUZ"
>
  <ul
    className="sc-gtssRu hoQdqF"
  >
    <li
      className="sc-dlnjPT jKgXLC"
    >
      <button
        className="sc-hKFyIo jaBECl"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-bars fa-w-14 "
          data-icon="bars"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </li>
    <li
      className="sc-dlnjPT jKgXLC"
    >
      <a
        href="/home"
      >
        Home
      </a>
    </li>
    <li
      className="sc-dlnjPT jKgXLC"
    >
      <a
        href="/patients"
      >
        Patients
      </a>
    </li>
  </ul>
</div>
`;
